### ISSUE TITLE
rollout in cycle with enabled rolloutRestartTargets

### ISSUE THREAD

**Issue by yotles** [hashicorp/vault-secrets-operator#582]
I've tried to use `VaultStaticSecret` to sync secrets from vault. And for proper rotating keys in secret I want to trigger deployment after updating secret in Vault. For this purpose I've added `RolloutRestartTriggered` to `VaultStaticSecret`. But even if I pinned version for secret kv-v2 my deployment triggered each minutes with logs `Rollout restart triggered for {Deployment ***}`.  If I add for example to description `RefreshAfter: 10s`, it will be triggered each 10s. 
At this time in logs of vault-secret-operator:
`
DEBUG	events	Rollout restart triggered for {Deployment ***}	{"type": "Normal", "object": {"kind":"VaultStaticSecret","namespace":"***","name":"***","uid":"***","apiVersion":"secrets.hashicorp.com/v1beta1","resourceVersion":"***"}, "reason": "RolloutRestartTriggered"}
DEBUG	events	Secret synced	{"type": "Normal", "object": {"kind":"VaultStaticSecret","namespace":"***","name":"***","uid":"***","apiVersion":"secrets.hashicorp.com/v1beta1","resourceVersion":"***"}, "reason": "SecretRotated"}`

To stop this behavior I've found only one solution, configure RefreshAfter for long period, for example 24h, in this way it works as expected, rollout restart happened only once when secret really changed with rotating version.

My VaultStaticSecret:

```yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: ***
  namespace: ***
spec:
  type: kv-v2
  mount: ***
  path: ***
  destination:
    name: ***
    create: true
  version: 4
  vaultAuthRef: ***
  rolloutRestartTargets:
    - kind: Deployment
      name: ***
```

**Expected behavior**
 
RolloutRestartTriggered happened only when I've changed version of secret in VaultStaticSecret. 

**Environment**
* Kubernetes version: v1.24.0
  * Distribution: OpenShift
* vault-secrets-operator version: 0.4.1

**Comment by yotles** [comment]
@benashz Do you know about this behavior, is it expected?

**Comment by bhaku** [comment]
@yotles We had the same problem. After updating to version `0.5.0` it went away, but remember to update the CRD!

**Comment by jameshwc** [comment]
Same issue here. We have deployments being rollout restarted every 30s (which is the `refreshAfter` value)
Here's the static secret yaml we have:
```yaml
 spec:
   destination:
     create: true
     name: secretkv
     overwrite: false
   hmacSecretData: true
   mount: ***/
   path: ***
   refreshAfter: 30s
   rolloutRestartTargets:
   - kind: Deployment
     name: ***
   type: kv-v2
   vaultAuthRef: static-auth
```

Updated to v0.5.1 didn't solve this issue.

**Comment by bhaku** [comment]
@jameshwc did you remember about the CRD update? They will not be updated when you update the version using helm.

**Comment by skeet70** [comment]
On version `0.5.2`, CRDs have been updated, but 

```yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultStaticSecret
metadata:
  name: argocd-secret
spec:
  vaultAuthRef: argocd
  mount: secret
  type: kv-v2
  path: argocd/argocd-secret
  destination:
    create: true
    overwrite: true
    name: argocd-secret
  rolloutRestartTargets:
  - kind: Deployment
    name: argocd-server
```

results in `argocd-server` coming up, working, and once confirmed working VSO triggers another rolling restart. This happens in an infinite ~1 min cycle.

**Comment by benashz** [comment]
@skeet70  - if you are using the rollout restart feature with static secrets, we recommend setting `spec.hmacSecretData=true`, that will ensure that only updated secret data triggers the rollout restart. In your example, the restarts will probably be triggered VSO restarting, since you are not specifying `spec.refreshAfter`.

**Comment by benashz** [comment]
Going to close this issue out. If you are still encountering a related problem, please open a new issue linking to this one. Thanks.
